### ISSUE TITLE
code: 98, kind: AddrInUse, message: "Address already in use

### ISSUE THREAD

**Issue by Pcummings** [Concordium/concordium-node#248]
**Bug Description**
I recently upgraded my node from version 1.0.1 to 3.0.1. Immediately after rebooting I got the following error and the node would not start. I finally followed the database invarian issue, and got it started again. Today I made my node a baker, and after restarting it was the same issue. I had to delete all but the -1 files in the database-v4. Now it has fully caught up, and shows on the dashboard as a baker and it seems like everything is operational, however I keep seeing this error on the log:
```Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + export LD_LIBRARY_PATH=/usr/local/lib
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + export LD_LIBRARY_PATH=/usr/local/lib
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + LD_LIBRARY_PATH=/usr/local/lib
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + ARGS=
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' -n '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: message repeated 22 times: [ + '[' -n '' ']']
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' -n 5000 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + ARGS=' --collect-interval 5000'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' -n https://dashboard.mainnet.concordium.software/nodes/post ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + ARGS=' --collect-interval 5000 --collector-url https://dashboard.mainnet.concordium.software/nodes/post'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' -n http://localhost:10000 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + ARGS=' --collect-interval 5000 --collector-url https://dashboard.mainnet.concordium.software/nodes/post --grpc-host http://localhost:10000'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' -n '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' -n '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' -n '' -a -f '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' collector == basic ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' collector == bootstrapper ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' collector == collector ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + '[' -n COLLECTOR_NODE_NAME ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19035]: + /usr/bin/node-collector --collect-interval 5000 --collector-url https://dashboard.mainnet.concordium.software/nodes/post --grpc-host http://localhost:10000 --node-name CCM-CONCORDIUM-NODE-1
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + LD_LIBRARY_PATH=/usr/local/lib
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n 8888 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n 5 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: message repeated 4 times: [ + '[' -n '' ']']
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + mkdir -p /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + mkdir -p /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + cd /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n bootstrap.mainnet.concordium.software:8888 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data --bootstrap-node bootstrap.mainnet.concordium.software:8888'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n 0.0.0.0 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data --bootstrap-node bootstrap.mainnet.concordium.software:8888 --rpc-server-addr 0.0.0.0'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n 10000 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data --bootstrap-node bootstrap.mainnet.concordium.software:8888 --rpc-server-addr 0.0.0.0 --rpc-server-port 10000'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n 10 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data --bootstrap-node bootstrap.mainnet.concordium.software:8888 --rpc-server-addr 0.0.0.0 --rpc-server-port 10000 --max-allowed-nodes 10'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n '--no-log-timestamp --bootstrapping-interval 1800 --no-dnssec --haskell-rts-flags=-N2' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data --bootstrap-node bootstrap.mainnet.concordium.software:8888 --rpc-server-addr 0.0.0.0 --rpc-server-port 10000 --max-allowed-nodes 10 --no-log-timestamp --bootstrapping-interval 1800 --no-dnssec --haskell-rts-flags=-N2'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: message repeated 5 times: [ + '[' -n '' ']']
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n 20 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data --bootstrap-node bootstrap.mainnet.concordium.software:8888 --rpc-server-addr 0.0.0.0 --rpc-server-port 10000 --max-allowed-nodes 10 --no-log-timestamp --bootstrapping-interval 1800 --no-dnssec --haskell-rts-flags=-N2 --hard-connection-limit 20'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: message repeated 3 times: [ + '[' -n '' ']']
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n 2 ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + ARGS=' --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data --bootstrap-node bootstrap.mainnet.concordium.software:8888 --rpc-server-addr 0.0.0.0 --rpc-server-port 10000 --max-allowed-nodes 10 --no-log-timestamp --bootstrapping-interval 1800 --no-dnssec --haskell-rts-flags=-N2 --hard-connection-limit 20 --thread-pool-size 2'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' -n '' -a -f '' ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + '[' basic == basic ']'
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19034]: + /usr/bin/concordium-node --listen-port 8888 --desired-nodes 5 --config-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config --data-dir /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data --bootstrap-node bootstrap.mainnet.concordium.software:8888 --rpc-server-addr 0.0.0.0 --rpc-server-port 10000 --max-allowed-nodes 10 --no-log-timestamp --bootstrapping-interval 1800 --no-dnssec --haskell-rts-flags=-N2 --hard-connection-limit 20 --thread-pool-size 2
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: [INFO  concordium_node::utils] Starting up concordium_node version 1.0.1!
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: [INFO  concordium_node::utils] Application data directory: /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/data
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: [INFO  concordium_node::utils] Application config directory: /var/lib/concordium/9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478/config
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: [INFO  concordium_node::utils] Network: enabled
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: [INFO  concordium_node::utils] Log level: info
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: [INFO  concordium_node::p2p::maintenance] My Node ID is 27ac52b95fbb1b8d
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: [INFO  concordium_node::p2p::maintenance] Listening on 172.31.12.98:8888
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: Error: Os { code: 98, kind: AddrInUse, message: "Address already in use" }
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: Could not listen on the given listen-port (8888).
Feb 14 20:00:30 ip-172-31-12-98 concordium-node-wrapper[19041]: Failed to create the node.```

I'm not sure how to fix this - could you perhaps help?

**Steps to Reproduce**

**Expected Result**

**Actual Result**

**Versions**
- Software Version - 3.0.1
- OS - Ubuntu 20.04
- Browser
- Mobile device, if applicable

**Comment by EmilLa1** [comment]
Hi, 

The error message indicates that the port is already in use. Perhaps you are trying to run two nodes on the same port?

**Comment by Pcummings** [comment]
Yes - I just don't understand why it would do that

**Comment by Pcummings** [comment]
Yeah @EmilLa1 you were right, the upgrade to 3.0.1 created new service files without deleting the old ones:
```
-rw-r--r-- 1 root root 1309 Sep  5 12:15 concordium-mainnet-node-collector.service
-rw-r--r-- 1 root root 2556 Sep  5 12:15 concordium-mainnet-node.service
-rw-r--r-- 1 root root 1248 May 31  2021 concordium-node-collector.service
-rw-r--r-- 1 root root 1870 May 31  2021 concordium-node.service
```
I kept the concordium-mainnet services and deleted the other ones, now there are no errors.

**Comment by abizjak** [comment]
@Pcummings to provide a bit of context. We did not expect anybody to upgrade from 1.0.0 to 3.*. The upgrade instructions from 1.0.0 to 1.1.3 did explicitly state to remove the old service (apt remove --purge concordium-node) and install the new one. Upgrades from 1.1.3 onwards will not need to remove old services, just install the new package.

The reason for the change was that we introduced two services so that uses can run both a testnet and mainnet node simultaneously without any dedicated configuration.
